frontend/src/__tests__/cypress/cypress/tests/mocked/projects/modelServingNim.cy.ts

```diff
@@ -0,0 +1,353 @@
+          .findComponent('overview')
+          .findByTestId('nvidia-nim-platform-card')
+          .should('contain', 'NVIDIA NIM model serving platform')
+          .and('contain', 'Models are deployed using NVIDIA NIM microservices.');
+
+        validateNvidiaNimModel(
+          projectDetails
+            .findComponent('overview')
+            .findByTestId('nvidia-nim-platform-card')
+            .findByTestId('model-serving-platform-button'),
+        );
+      });
+
+      it("should allow deploying NIM from a Project's Models tab when the only platform", () => {
+        initInterceptsToEnableNim({});
+        projectDetails.visitSection('test-project', 'model-server');
+        projectDetails.shouldBeEmptyState('Models', 'model-server', true);
+        projectDetails.findServingPlatformLabel().should('exist');
+
+        cy.contains('Start by adding a model server');
+        cy.contains(
+          'Model servers are used to deploy models and to allow apps to send requests to your models. Configuring a model server includes specifying the number of replicas being deployed, the server size, the token authentication, the serving runtime, and how the project that the model server belongs to is accessed.',
+        );
+
+        const deployButton = projectDetails
+          .findComponent('model-server')
+          .findByTestId('deploy-button');
+        validateNvidiaNimModel(deployButton);
+      });
+
+      it("should allow deploying NIM from a Project's Models tab when multiple platforms exist", () => {
+        initInterceptorsValidatingNimEnablement({
+          disableKServe: false,
+          disableModelMesh: false,
+          disableNIMModelServing: false,
+        });
+
+        projectDetails.visitSection('test-project', 'model-server');
+        projectDetails.shouldBeEmptyState('Models', 'model-server', true);
+        projectDetails.findServingPlatformLabel().should('not.exist');
+
+        projectDetails.findSingleModelDeployButton().should('exist');
+        projectDetails.findMultiModelButton().should('exist');
+
+        findNimModelServingPlatformCard()
+          .should('contain', 'Models are deployed using NVIDIA NIM microservices.')
+          .and('contain', 'NVIDIA NIM model serving platform');
+
+        validateNvidiaNimModel(findNimModelDeployButton());
+      });
+
+      it('should display an error when failed to fetch NIM model list', () => {
+        initInterceptsToEnableNim({});
+        const componentName = 'overview';
+        projectDetails.visitSection('test-project', componentName);
+        const overviewComponent = projectDetails.findComponent(componentName);
+        overviewComponent.should('exist');
+        const deployModelButton = overviewComponent.findByTestId('model-serving-platform-button');
+        deployModelButton.should('exist');
+        deployModelButton.click();
+        cy.contains('There was a problem fetching the NIM models. Please try again later.');
```

Comment:
We should probably scope checks for content to specific areas -- more specific text probably has less of a problem, but it seems a bit lacking in the page object department this way